tree/ntuple/v7/inc/ROOT/RNTupleUtil.hxx

```diff
@@ -108,17 +109,43 @@ public:
    ClusterSize_t::ValueType GetIndex() const { return fIndex; }
 };
 
+/// RNTupleLocator payload that is common for object stores using 64bit location information.
+/// This might not contain the full location of the content. In particular, for page locators this information may be
+/// used in conjunction with the cluster and column ID.
+struct RNTupleLocatorObject64 {
```

Comment:
Since DAOS is using the std::uint64_t variant, we can just remove `RNTupleLocatorObject64`, no?

Reply:
Yes and no.  The DAOS backend is currently (wrongly) using the on-file locators; changing that is a follow-up PR, to avoid more changes in this PR.

Also, it is true that `RNTupleLocatorObject64` is a POD type containing only a `std::uint64_t` and we might use the `std::uint64_t` alternative of the variant.  However, I'm in favor of keeping it given that the code emitted by the compiler should be equivalent, and due to the following two reasons:

1. I find it more informational to have
```c++
locator.fPosition = RNTupleLocatorObject64{42U};
locator.fType = RNTupleLocator::kTypeDAOS;
```
as opposed to
```c++
locator.fPosition = 42U;
locator.fType = RNTupleLocator::kTypeDAOS;
```
2. It is also safer: the following two things should throw during serialization due to `std::get<T>()` trying to access the wrong type:
```c++
locator.fPosition = RNTupleLocatorObject64{42U};
locator.fType = RNTupleLocator::kTypeFile;
// or
locator.fPosition = 42U;
locator.fType = RNTupleLocator::kTypeDAOS;
```

Reply:
Fair enough.